interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: After researching I found that there's 2 types of parameters(parameters are also considered as variables). The input which pass in values to your function, and the output which return multiple values from your function.
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: .map() is a higher order function you can only use on arrays and are used for iteration. Their predefined parameters are value, index, and array in that order. the parameters are part of the callback function passed through .map(). index and array are optional parameters as you can just declare value. Also note that .map() returns a new array, if you're looking to iterate through an array and change your existing array use .forEach()
 
-  Researched answer:
+  Researched answer: Looks like I was pretty close to the entire answer but it seems I missed thisArg which is the value to use this when you are executing the callback function. callback functions also only applies to indexes of the array that actually have values assigned to them, undefined counts here as a value.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Both .map() and .filter() are used to iterate through arrays, take in callback functions with value(index, and array are optional here too), and both also create new arrays once returned. The primary difference between the two is map iterates through the array where filter iterates through the array and contains a built in conditional that takes in values and returns only the ones that come back as matching your conditionals requirements. Effectively "filtering" the data that you return. Imagine using for loops with conditional statements... yuck
 
-  Researched answer:
+  Researched answer: When using map you iterate through the array and have it return a whole new array with changed values based on what you want to change, however with filter it just checks if your array meets certain requirements and makes decisions from there and won't alter the values of your array. 
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: First what is a function and a method. A function is a reusable dynamic code block assigned to a variable. A method is similar, being a code block that can be applied to classes and other functions. So the difference would be methods have to be appended to a parent component and functions can be used seperately from other functions. 
 
-  Researched answer:
+  Researched answer: Methods take in function definitions and are applied to objects. Methods are also stored in objects as properties that can be assigned to classes. Objects that take on a method don't need parentheses like how functions do when calling them.
```

Comment:
Great! Take a moment to share a time you used this tool to segway into talking about your projects!